sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -2057,14 +2055,17 @@ case class ValidateExternalType(child: Expression, expected: DataType, externalD
     if (checkType(input)) {
       input
     } else {
-      throw new RuntimeException(s"${input.getClass.getName}$errMsg")
+      throw new SparkRuntimeException("INVALID_EXTERNAL_TYPE",
```

Comment:
Please, create the exception in `QueryExecutionErrors`, and invoke the method from codegen and non-codegen.